Eastern-Research-Group/mywaterway
# Bugfix/611 duplicated monitoring locations

## Related Issues:

- [HMW-611](https://jira.epa.gov/browse/HMW-611)

## Main Changes:

- Addressed a race condition that occurs when using the `applyEdits` method of Esri's `FeatureLayer`.
  - I initially attempted to adapt the solution from the workaround provided in the Esri forums, https://community.esri.com/t5/arcgis-javascript-maps-sdk-questions/arcgis-api-4-17-client-side-feature-layer/td-p/59530, which seemed to work occasionally, but not consistently. Instead, I switched to adding the `periodOfRecord` data to monitoring locations by mutating the fetched data, rather than dispatching a new object. This avoids the race condition, but doesn't address the underlying issue, so I'll keep a note of the initial attempts in case this crops up again in the future.

## Steps To Test:

1. Navigate to http://localhost:3000/community/dc/overview and toggle on the monitoring locations.
2. On the map, confirm that there is only one feature for each monitoring location.
3. Refresh the page, then repeat step 2.
4. Navigate to a Tribal page by clicking, then repeat step 2.
5. Confirm the characteristics filters and date slider on the Community page's Monitoring tab still function as they should.
